src/Covenant/Expr.hs

```diff
@@ -413,4 +459,7 @@ toIdList = \case
 refToId :: Ref -> Maybe Id
 refToId = \case
   AnArg _ -> Nothing
-  AnId i -> Just i
+  AnId (TypedId i _) -> Just i
+
+pushScope :: forall (n :: Natural). ExprType -> Scope n -> Scope (n + 1)
+pushScope ty (Scope tys) = Scope (ty : tys)
```

Comment:
@kozross `pushScope` for your reference.

Reply:
Ah, so essentially, `Scope n` also stores exactly `n` types for its arguments. Seems like a good compromise.